pilot/pkg/model/config.go

```diff
@@ -1003,6 +1033,82 @@ func (store *istioConfigStore) AuthenticationPolicyByDestination(hostname string
 	return &out
 }
 
+// Set jwks_uri through openID discovery if it's not set in auth policy.
+func (c *authCache) SetAuthenticationPolicyJwksURIs(policy *authn.Policy) {
+	if policy == nil {
+		return
+	}
+
+	for _, method := range policy.Peers {
+		switch method.GetParams().(type) {
+		case *authn.PeerAuthenticationMethod_Jwt:
+			policyJwt := method.GetJwt()
+			uri, err := c.getJwksURI(policyJwt)
+			if err != nil {
+				log.Warnf("Failed to get jwks_uri for issuer %q: %v", policyJwt.Issuer, err)
+				continue
+			}
+			policyJwt.JwksUri = uri
+		}
+	}
+	for _, method := range policy.Origins {
+		// JWT is only allowed authentication method type for Origin.
+		policyJwt := method.GetJwt()
+		uri, err := c.getJwksURI(policyJwt)
+		if err != nil {
+			log.Warnf("Failed to get jwks_uri for issuer %q: %v", policyJwt.Issuer, err)
+			continue
+		}
+		policyJwt.JwksUri = uri
+	}
+}
+
+// Get jwks_uri through openID discovery if it's not set in auth policy, and cache the jwks_uri for furture use.
+func (c *authCache) getJwksURI(policyJwt *authn.Jwt) (string, error) {
```

Comment:
It's cleaner that function take in only issuer string and return jwksURI. It can also be renamed to resolveJwksURIUsingOpenID.

Reply:
done.